config/sets/symfony/symfony50.php

```diff
@@ -2,16 +2,28 @@
 
 declare(strict_types=1);
 
+use PHPStan\Type\StringType;
 use Rector\Config\RectorConfig;
 use Rector\Renaming\Rector\MethodCall\RenameMethodRector;
 use Rector\Renaming\Rector\Name\RenameClassRector;
 use Rector\Renaming\ValueObject\MethodCallRename;
+use Rector\TypeDeclaration\Rector\ClassMethod\AddParamTypeDeclarationRector;
+use Rector\TypeDeclaration\ValueObject\AddParamTypeDeclaration;
 
 # https://github.com/symfony/symfony/blob/5.0/UPGRADE-5.0.md
 
 return static function (RectorConfig $rectorConfig): void {
     $rectorConfig->import(__DIR__ . '/symfony50-types.php');
 
+    $rectorConfig->ruleWithConfiguration(AddParamTypeDeclarationRector::class, [
+        new AddParamTypeDeclaration(
+            'Symfony\Component\Security\Core\User\UserProviderInterface',
+            'loadUserByUsername',
+            0,
+            new StringType(),
+        ),
+    ]);
```

Comment:
per your link, should it be on symfony 5.4 set?

Reply:
My links where incorrect i updated them sorry for the confusion: https://github.com/rectorphp/rector-symfony/pull/699#issue-2818539804

i also added some more that were missing